airflow/jobs/scheduler_job.py

```diff
@@ -596,13 +597,24 @@ def _process_executor_events(self, session: Session = None) -> int:
                 )
                 self.log.error(msg, ti, state, ti.state, info)
 
+                try:
+                    get_dag(self.subdir, ti.dag_id)
```

Comment:
It's not clear to me why we even need this? What's going on here?

Reply:
When there's an error in parsing the dag, maybe due to resource constraints, the task callback is not run even though it's sent to the processor. The task gets stuck in queued. 
But when there's db operational error while trying to execute a task, it results in this error but the task retries.
One other thing we can do is add a message to inform the user that if there's a resource constraint that the task can get stuck. Then remove parsing of dags